### ISSUE TITLE
macOS Catalina compatibility issues

### ISSUE THREAD

**Issue by mparksUCO** [popgenmethods/smcpp#139]
Hello,
I am having problems installing smc++ on my mac laptop and desktop, and it seems like there may be compatibility issues caused by the Catalina OS - particularly between gcc and Xcode. For example, here is an error message I get during the installation process:

*******
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/execinfo.h:42:49: error: expected initializer before '__OSX_AVAILABLE_STARTING'
 void backtrace_symbols_fd(void* const*,int,int) __OSX_AVAILABLE_STARTING(__MAC_10_5, __IPHONE_2_0);
                                                 ^~~~~~~~~~~~~~~~~~~~~~~~
In file included from include/common.h:16:0,
                 from include/jcsfs.h:6,
                 from src/jcsfs.cpp:4:
include/stacktrace.h: In function 'void print_stacktrace(FILE*, unsigned int)':
include/stacktrace.h:21:71: error: 'backtrace' was not declared in this scope
     int addrlen = backtrace(addrlist, sizeof(addrlist) / sizeof(void*));
                                                                       ^
include/stacktrace.h:30:60: error: 'backtrace_symbols' was not declared in this scope
     char** symbollist = backtrace_symbols(addrlist, addrlen);
                                                            ^
error: command 'gcc-6' failed with exit status 1
*******

Are other users having similar issues, if so is there a simple way to get around this? Apologies if I am missing something obvious, but I have been following the installation instructions both for Anaconda and a local build/installation and am not having any luck getting things up and running.

Thanks in advance for any help you can offer, I am happy to provide any more information that might be helpful.

**Comment by terhorst** [comment]
Thanks for reporting. I don't know anything about this error other than what I found on Google. It seems possibly related to Homebrew/homebrew-core#44579. From that, my best guess is that it's a compiler configuration issue. The software builds on my MacBook using gcc-8, so one recommendation might be to upgrade your compiler. In general, I have found that building software using the GCC toolchain has gotten progressively harder with each new OS X release.
